montecarlo_localization.py

```diff
@@ -351,12 +355,16 @@ def new_pose_from_log_delta(self, new_log_pose):
         log_delta_theta = new_log_pose[2] - self.prev_log_pose[2]
         # Fwd motion in log frame == Fwd motion in particle framea
         fwd_motion = math.sqrt(log_delta_x**2 + log_delta_y**2)
+        motion_heading = math.atan2(log_delta_y, log_delta_x)
+        
         # Calculate and add stochastic theta and forward error
         new_theta_error = log_delta_theta * self.sigma_theta_pct * np.random.normal()
         new_current_theta = self.pose[2] + log_delta_theta + new_theta_error
         # Wrap radians to enforce range 0 to 2pi
         new_current_theta = new_current_theta % (2*np.pi)
-
+        heading_vs_direction_angle = abs(motion_heading-new_current_theta+np.pi) % (2*np.pi)-np.pi
+        if (abs(heading_vs_direction_angle) > (np.pi/2)):
```

Comment:
superfluous parentheses and whitespace required:
```suggestion
        if abs(heading_vs_direction_angle) > (np.pi / 2):
```